src/pages/iou/request/step/IOURequestStepAmount.js

```diff
@@ -154,21 +164,40 @@ function IOURequestStepAmount({
         Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID));
     };
 
+    const saveAmountAndCurrency = ({amount}) => {
+        const newAmount = CurrencyUtils.convertToBackendAmount(Number.parseFloat(amount));
+
+        // If the value hasn't changed, don't request to save changes on the server and just close the modal
+        if (newAmount === TransactionUtils.getAmount(transaction) && currency === TransactionUtils.getCurrency(transaction)) {
+            Navigation.dismissModal();
+            return;
+        }
+
+        if (isSplitBill) {
+            IOU.setDraftSplitTransaction(transactionID, {amount: newAmount, currency});
+            Navigation.goBack(backTo);
+            return;
+        }
+
+        IOU.updateMoneyRequestAmountAndCurrency(transactionID, reportID, currency, newAmount);
```

Comment:
In the creating flow, we only should set new value to ONYX instead of call API to BE